qiskit/providers/aer/noise/errors/quantum_error.py

```diff
@@ -292,367 +408,69 @@ def to_dict(self):
-            zip(combined_noise_circuits, combined_noise_probabilities),
-            self.number_of_qubits + other.number_of_qubits)
-        return QuantumError(noise_ops)
+        circs = [self._compose_circ(lqc, rqc, qubits=qargs, front=front)
+                 for lqc in self.circuits
+                 for rqc in other.circuits]
+        probs = [lpr * rpr
+                 for lpr in self.probabilities
+                 for rpr in other.probabilities]
+        return QuantumError(zip(circs, probs))
 
     @staticmethod
-    def _combine_kraus(noise_ops, num_qubits):
-        """Combine any noise circuits containing only Kraus instructions."""
-        kraus_instr = []
-        kraus_probs = []
-        new_circuits = []
-        new_probs = []
-        # Partion circuits into Kraus and non-Kraus
-        for circuit, prob in noise_ops:
-            if len(circuit) == 1 and circuit[0]['name'] == 'kraus':
-                kraus_instr.append(circuit[0])
-                kraus_probs.append(prob)
-            else:
-                new_circuits.append(circuit)
-                new_probs.append(prob)
-        # Combine matching Kraus instructions via Choi rep
-        if len(kraus_probs) == 1:
-            new_circuits.append([kraus_instr[0]])
-            new_probs.append(kraus_probs[0])
-        elif len(kraus_probs) > 1:
-            dim = 2 ** num_qubits
-            iden = SuperOp(np.eye(dim ** 2))
-            choi_sum = Choi(np.zeros((dim ** 2, dim ** 2)))
-            for prob, instr in zip(kraus_probs, kraus_instr):
-                choi_sum = choi_sum + prob * iden.compose(Kraus(instr['params']),
-                                                          instr['qubits'])
-            # Renormalize the Choi operator to find probability
-            # of Kraus error
-            chan_prob = abs(np.trace(choi_sum.data) / dim)
-            chan_instr = {
-                "name": "kraus",
-                "qubits": list(range(num_qubits)),
-                "params": Kraus(choi_sum / chan_prob).data
-            }
-            new_circuits.append([chan_instr])
-            new_probs.append(chan_prob)
-        return list(zip(new_circuits, new_probs))
+    def _enlarge_qreg(qc: QuantumCircuit, num_qubits: int):
+        if qc.num_qubits < num_qubits:
+            enlarged = QuantumCircuit(num_qubits)
+            return enlarged.compose(qc)
+        return qc
 
     @staticmethod
-    def _check_instr(name):
-        """Check if instruction name can be converted to standard operator"""
-        return name in [
-            'kraus', 'unitary', 'reset', 'u1', 'u2', 'u3', 'id', 'x', 'y', 'z',
-            'h', 's', 'sdg', 't', 'tdg', 'cx', 'cz', 'swap', 'ccx'
-        ]
+    def _compose_circ(lqc: QuantumCircuit, rqc: QuantumCircuit, qubits, front):
+        if lqc.num_qubits < rqc.num_qubits:
+            lqc = QuantumError._enlarge_qreg(lqc, rqc.num_qubits)
+        return lqc.compose(rqc, qubits=qubits, front=front)
 
```

Comment:
It might depend on the future implementation of `QuantumCircuit.compose()` (it's not yet implemented!) but so far `QuantumCircuit.compose()` requires lqc (self) is larger than rqc (other), so I think there is no problem for now.

Reply:
I found the newly introduced `qargs` argument makes things difficult.
If we didn't have that, we just swap lqc and rqc if `front=True` and enlarge lqc, then compose them.
However, `qargs` must be a qubit subset of `self` (lqc) and its length must be the same as `other.num_qubits`. Consequently, `self.num_qubits < other.num_qubits` and hence we cannot enlarge `self` (lqc) within this function.
I've updated the code based on the above findings (782038e).